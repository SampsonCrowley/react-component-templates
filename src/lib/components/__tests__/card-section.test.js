import React from 'react'
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';

import CardSection from '../card-section'

describe('Components - CardSection', () => {
  const div = document.createElement('div');

  const createCardSection = ({...props}) => {
    ReactDOM.render((
      <Router>
        <CardSection {...props} />
      </Router>
    ), div);
    return div.querySelector('section')
  }

  it('renders an section tag', () => {
    const rendered = createCardSection()
    expect(rendered).toBeTruthy()
    expect(rendered.tagName).toBe("SECTION")
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders header and main tags', () => {
    const rendered = createCardSection()
    expect(rendered.querySelector('header')).toBeTruthy()
    expect(rendered.querySelector('main')).toBeTruthy()
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders an H3 tag in the header', () => {
    const rendered = createCardSection()
    expect(rendered.querySelector('header h3')).toBeTruthy()
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders children', () => {
    expect(createCardSection({
      children: 'Test'
    }).querySelector('main').innerHTML).toMatch(/^Test$/)
    ReactDOM.unmountComponentAtNode(div);
  })

  it('is snapshotable', () => {
    const tree = renderer
      .create(
        <Router>
          <CardSection to='/'>Test</CardSection>
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot()
  })
})
