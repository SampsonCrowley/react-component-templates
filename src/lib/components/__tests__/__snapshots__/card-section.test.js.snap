// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - CardSection is snapshotable 1`] = `
<section
  className="card text-default "
  to="/"
>
  <header
    className="text-center card-header "
  >
    <h3
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
    />
    
  </header>
  <div
    className="card-body "
  >
    Test
  </div>
</section>
`;
